.github/workflows/dependency-upgrade.yaml

```diff
@@ -0,0 +1,72 @@
+---
+name: Dependency Upgrade Check
+
+on:
+  schedule:
+    # Runs daily at noon UTC (I guess)
+    - cron:  '0 12 * * *'
+
+jobs:
+  check-upgrades:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dependency: ["Stargate", "Medusa", "Reaper"]
+        include:
+          - dependency: Stargate
+            version_path: ".stargate.version"
+            docker_hub_url: "https://registry.hub.docker.com/v2/repositories/stargateio/stargate-3_11/tags?page_size=1"
+          - dependency: Medusa
+            version_path: ".medusa.image.tag"
+            docker_hub_url: "https://registry.hub.docker.com/v2/repositories/k8ssandra/medusa/tags?page_size=25"
+          - dependency: Reaper
+            version_path: ".reaper.image.tag"
+            docker_hub_url: "https://registry.hub.docker.com/v2/repositories/thelastpickle/cassandra-reaper/tags?page_size=25"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Update PATH
+        run: |
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+      - name: Install YQ and JQ
+        run: |
+          scripts/install-yq.sh v4.6.3
```

Comment:
I'm wondering since the version references and other value constants are increasing across the .yaml files under the `workflows` directory, if it would be beneficial to have them defined in one place?  Something similar to the approach we are thinking about with `image` repositories.

Reply:
Very good point!
I created a script that sets the versions in env variables and can be called from the different workflows.